### PR TITLE
Add Concat op.

### PR DESCRIPTION
**Pull request by robinvanemden**
All ONNX backend tests pass. First op that contains variadic inputs.

### CODE REVIEW

File: src/nodes/concat.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        std::vector<const Tensor *> node_inputs;
+        const Tensor *output;
+
+        void print_parameters(std::ostream &dst, bool decorate) const override {
+            size_t input_count = node_inputs.size();
+            for (size_t i = 0; i < input_count; i++) {
+                node_inputs[i]->print_tensor(dst, !decorate);
+                if (i < (input_count - 1))
+                    dst << ", ";
+            }
```
Comment (kraiskil) [inline src/nodes/concat.h:21]: The output tensor needs to be printed

**Review by kraiskil — CHANGES_REQUESTED** [review]
(first time doing a review on github, not sure what this box does now :)

**Comment by kraiskil** [comment]
Thanks for the PR.

I noticed that in the generated test case the concat node lacks the output tensor given as a parameter. The tests still pass, since there is a global reserved tensor of the same name (it is intended that a pointer to this is passed in as a parameter).

There are some pytorch-generated tests in the backend test suite too, one of which tests concat - and this one fails:
`ONNX_type_test(operator_concat2 ${ONNX_BACKEND_TEST_DATA_DIR}/pytorch-operator/test_operator_concat2 ONNX_backend_pytorch_conv 0.00001 0)`

**Comment by kraiskil** [comment]
Also, the generated code lacks some indentations, which make it a bit difficult to read.
Tabs are preferred, since that allows the reader to adjust the indentation depth to their preference, and keeps the indentation consistent across nodes.

File: src/nodes/concat.h (hunk 1 above)
Comment (robinvanemden) [reply src/nodes/concat.h:21]: Indeed! I missed that one - doesn't interfere with the backend tests. I will fix it asap!

**Comment by robinvanemden** [comment]
Thanks for the feedback! I will make the suggested changes asap.

File: src/nodes/concat.h (hunk 1 above)
Comment (kraiskil) [reply src/nodes/concat.h:21]: Yup. I would even say not catching this is a bug in the backend test generator...

**Comment by robinvanemden** [comment]
I made the requested changes and cleaned up the code some more. Thanks again for the review!

**Pull request merged**